### PR TITLE
First pass of implementation for Aliasable Resources

### PR DESCRIPTION
**Pull request by rocktavious**
Resolves https://github.com/OpsLevel/team-platform/issues/496

### Problem

Currently we have no unified way to handle Aliasable resources like we do for Taggable ones.  

### Solution

Apply the same interface pattern that was done to Taggable resources but for Aliasable ones.

### Checklist

- [x] I have run this code, and it appears to resolve the stated issue.
- [ ] This PR does not reduce total test coverage
- [x] This PR has no user interface changes or has already received approval from product management to change the interface.
- [x] Does this change require a Terraform schema change?
  - If so what is the ticket or PR # TBD
- [x] Make a [changie](https://github.com/OpsLevel/opslevel-go/blob/main/CONTRIBUTING.md#changie-change-log-generation) entry that explains the customer facing outcome of this change

### CODE REVIEW

File: aliases.go [hunk 1]
```diff
@@ -2,8 +2,47 @@ package opslevel
...
+	"fmt"
 )
 
+type AliasableResourceInterface interface {
+	GetAliases() []string
+	ResourceId() ID
+	AliasableType() AliasOwnerTypeEnum
+}
+
+func (client *Client) GetAliasableResource(resourceType AliasOwnerTypeEnum, identifier string) (AliasableResourceInterface, error) {
```
Comment (rocktavious) [inline aliases.go:14]: This is the meat and potatoes function that will be tested

File: aliases_test.go [hunk 2]
```diff
@@ -137,3 +137,27 @@ func TestDeleteAliasNotFound(t *testing.T) {
...
+		),
+		autopilot.NewTestRequest(
+			``,
+			`{}`,
+			`{}`,
+		),
+	}
+	client := BestTestClient(t, "tags/get_aliasable_resource", requests...)
+	// Act
+	service, err1 := client.GetAliasableResource(ol.AliasOwnerTypeEnumService, "")
```
Comment (rocktavious) [inline aliases_test.go:156]: With this setup i should beable to test each branch of the switch statement in 1 go

File: tags.go [hunk 3]
```diff
@@ -77,13 +77,10 @@ func (client *Client) GetTaggableResource(resourceType TaggableResource, identif
 	case TaggableResourceUser:
 		taggableResource, err = client.GetUser(identifier)
 	default:
-		return nil, fmt.Errorf("not a taggable resource type: %s", string(resourceType))
+		err = fmt.Errorf("not a taggable resource type: %s", resourceType)
```
Comment (rocktavious) [inline tags.go:80]: This just cleans up the logic a bit - i used the same pattern in GetAliasableResource

File: clientGQL_test.go [hunk 4]
```diff
@@ -128,7 +132,18 @@ func (testDataTemplater *TestDataTemplater) ParseTemplatedString(contents string
 	return strings.TrimSpace(data.String())
 }
 
+func stripWhitespace(input string) string {
```
Comment (rocktavious) [inline clientGQL_test.go:135]: Adding this ensures that we can have readable templates - see https://github.com/OpsLevel/opslevel-go/pull/483/commits/7fb2cddcbd5495177af4e3a899fb83849a81d185

File: team.go [hunk 5]
```diff
@@ -203,6 +207,11 @@ func (team *Team) GetTags(client *Client, variables *PayloadVariables) (*TagConn
 	return team.Tags, nil
 }
 
+func (team *Team) GetAliases() []string {
+	// TODO: should this return Aliases or ManagedAliases ???
```
Comment (rocktavious) [inline team.go:211]:
```suggestion
```

**Review by davidbloss — APPROVED** [review]
LGTM 🎉

**Pull request merged**